### PR TITLE
[11.0][MIG] product_supplierinfo_for_customer

### PR DESCRIPTION
**Pull request by MiquelRForgeFlow**
Superseeds https://github.com/OCA/product-attribute/pull/367 to update commit history of v10.

@sergio-teruel I know that maybe I should have asked first, but I wouldn't want to bother you or giving you more work to do. I have maintained all your changes except two:

- readme file: You deleted content of readme, but @pedrobaeza says that is better to keep it, so I kept it when adding readme folder.
- product_view.xml: When you did the migration, the v10 at that moment was doing "extended" versions of views, and your migration improved that. But later in v10, it was abandoned that approach and taken approach of create new views (non inheriting customer views). I maintained that approach.

### CODE REVIEW

File: product_supplierinfo_for_customer/__manifest__.py
Comment (elicoidal) [inline product_supplierinfo_for_customer/__manifest__.py:1]: probably need to update the dates as you are modifying at present

File: product_supplierinfo_for_customer/demo/product_demo.xml
Comment (elicoidal) [inline product_supplierinfo_for_customer/demo/product_demo.xml:4]: add copyright

File: product_supplierinfo_for_customer/views/product_view.xml
Comment (elicoidal) [inline product_supplierinfo_for_customer/views/product_view.xml:6]: copyright

File: product_supplierinfo_for_customer/views/product_view.xml
Comment (elicoidal) [inline product_supplierinfo_for_customer/views/product_view.xml]: Please improve the readbility of the xml with one attribute per line

**Review by elicoidal — CHANGES_REQUESTED** [review]
some details

**Comment by thomaspaulb** [comment]
@elicoidal Have your reviews been successfully addressed?

File: product_supplierinfo_for_customer/i18n/de.po
Comment (thomaspaulb) [inline product_supplierinfo_for_customer/i18n/de.po:128]: Quite a number of translations are for attributes that don't need translation, such as [this one](https://github.com/OCA/product-attribute/pull/378/files#diff-3f11b17f61198057bc5aca5ce4810fd8R131). Could you re-export the German translation to see whether they disappear, or otherwise, add `t-translation="off"` ([link](https://github.com/OCA/OCB/blob/11.0/odoo/tools/translate.py#L211)) to the relevant attributes.

**Review by thomaspaulb — APPROVED** [review]
Translation fix is optional IMO, the tests run well so I think this can be merged.

File: product_supplierinfo_for_customer/__manifest__.py
Comment (mpanarin) [inline product_supplierinfo_for_customer/__manifest__.py]: non-blocking but i find this line to be redundant.

The `installable:False` was not used from migrations to v 9.0

File: product_supplierinfo_for_customer/models/product_product.py
Comment (mpanarin) [inline product_supplierinfo_for_customer/models/product_product.py]: i think you can `assert` this in order to keep it flatter

File: product_supplierinfo_for_customer/models/product_product.py
Comment (mpanarin) [inline product_supplierinfo_for_customer/models/product_product.py]: :stop_sign:  you are overwriting the dict on every iteration
as well as you will only have one iteration, as you return `prices` inside the cycle

File: product_supplierinfo_for_customer/models/product_supplierinfo.py
Comment (mpanarin) [inline product_supplierinfo_for_customer/models/product_supplierinfo.py]: this is broken due to multi, and will fail if you have recordset of >1 values

Also, as you use this onchange only in tests it is better to leave a comment about that in the method as it is confusing )

File: product_supplierinfo_for_customer/tests/test_product_supplierinfo_for_customer.py
Comment (mpanarin) [inline product_supplierinfo_for_customer/tests/test_product_supplierinfo_for_customer.py]: Consider changing this to SavepointCase. As you have a lot of test cases it will be a lot quicker

**Review by mpanarin — CHANGES_REQUESTED** [review]

**Review by elicoidal — APPROVED** [review]
LGTM

File: product_supplierinfo_for_customer/views/product_view.xml
Comment (SimoRubi) [inline product_supplierinfo_for_customer/views/product_view.xml:102]: Why did you remove `domain="[('type','=','supplier')]"`?

File: product_supplierinfo_for_customer/views/product_view.xml
Comment (SimoRubi) [inline product_supplierinfo_for_customer/views/product_view.xml:141]: Why did you remove `'default_product_tmpl_id':id`?

File: product_supplierinfo_for_customer/views/product_view.xml
Comment (SimoRubi) [inline product_supplierinfo_for_customer/views/product_view.xml:158]: Why did you remove `'default_product_tmpl_id':id`?

File: product_supplierinfo_for_customer/views/product_view.xml
Comment (SimoRubi) [inline product_supplierinfo_for_customer/views/product_view.xml:84]: Why did you remove `'default_product_tmpl_id':id`?

File: product_supplierinfo_for_customer/models/product_product.py
Comment (SimoRubi) [inline product_supplierinfo_for_customer/models/product_product.py]: Ok, you want to return the first supplierinfo that you find. But what if you find more than one? I suggest adding `limit=1` to the search arguments

File: product_supplierinfo_for_customer/models/product_product.py
Comment (SimoRubi) [inline product_supplierinfo_for_customer/models/product_product.py]: Returning from inside the loop doesn't seem to be a good idea unless you want to return the _first you find_ like in `_get_price_from_supplierinfo`.
I suggest to get the prices from the `super(ProductProduct, self).price_compute(price_type, uom, currency, company)`; then overwrite the result with what you find for the supplier
and then overwrite the entries for which `_get_price_from_supplierinfo` returns something different from `0.0`.

**Review by SimoRubi — CHANGES_REQUESTED** [review]
Thanks for your contribution!

File: product_supplierinfo_for_customer/README.rst
Comment (tarteo) [inline product_supplierinfo_for_customer/README.rst:1]: Can you generate the readme using https://github.com/OCA/maintainer-tools/blob/master/tools/gen_addon_readme.py

File: product_supplierinfo_for_customer/models/product_supplierinfo.py
Comment (tarteo) [inline product_supplierinfo_for_customer/models/product_supplierinfo.py]: string here is not needed Odoo automatically capitalizes field names.

File: product_supplierinfo_for_customer/models/product_product.py
Comment (luistorresm) [reply product_supplierinfo_for_customer/models/product_product.py]: What do you think about change the logic here:

```
        if not partner_id:
             return 0
        supplierinfo = self.env['product.supplierinfo'].search(
           ['|', ('product_tmpl_id', '=', product.product_tmpl_id.id),
            ('product_id', '=', product.id),
            ('type', '=', 'customer'),
            ('name', '=', partner_id)])
        return supplierinfo.price
```

Whit this you gain a indentation

And maybe add a `self.onsere_one()` in the method, you has the return in the `for`

**Review by tarteo — CHANGES_REQUESTED** [review]
Just some minor comments (besides the suggestions already made by the other reviewers)

**Comment by rafaelbn** [comment]
https://github.com/OCA/product-attribute/pull/367

**Comment by sergio-teruel** [comment]
@mreficent Can you generate the new readme?

**Comment by MiquelRForgeFlow** [comment]
Thank you for all the reviews. I will do the changes in the next few days.

**Comment by pedrobaeza** [comment]
@mreficent can you please finish this?

File: product_supplierinfo_for_customer/tests/test_product_supplierinfo_for_customer.py
Comment (MiquelRForgeFlow) [reply product_supplierinfo_for_customer/tests/test_product_supplierinfo_for_customer.py]: I suppose this is an optional change, right?

File: product_supplierinfo_for_customer/models/product_supplierinfo.py
Comment (MiquelRForgeFlow) [reply product_supplierinfo_for_customer/models/product_supplierinfo.py]: I will just delete `@api.multi`

File: product_supplierinfo_for_customer/tests/test_product_supplierinfo_for_customer.py
Comment (pedrobaeza) [reply product_supplierinfo_for_customer/tests/test_product_supplierinfo_for_customer.py]: Well, it's a well received optimization. The problem both have is that when testing on parallel with other modules that tests reports, both will hang, so `HttpCase` is the only that saves that problem.

File: product_supplierinfo_for_customer/README.rst
Comment (MiquelRForgeFlow) [reply product_supplierinfo_for_customer/README.rst:1]: no, I can't :(

File: product_supplierinfo_for_customer/views/product_view.xml
Comment (MiquelRForgeFlow) [reply product_supplierinfo_for_customer/views/product_view.xml:102]: error, not intended

File: product_supplierinfo_for_customer/views/product_view.xml
Comment (MiquelRForgeFlow) [reply product_supplierinfo_for_customer/views/product_view.xml:141]: is not removed

File: product_supplierinfo_for_customer/views/product_view.xml
Comment (MiquelRForgeFlow) [reply product_supplierinfo_for_customer/views/product_view.xml:158]: is not removed

File: product_supplierinfo_for_customer/views/product_view.xml
Comment (MiquelRForgeFlow) [reply product_supplierinfo_for_customer/views/product_view.xml:84]: is not removed

File: product_supplierinfo_for_customer/i18n/de.po
Comment (MiquelRForgeFlow) [reply product_supplierinfo_for_customer/i18n/de.po:128]: I won't touch translations :S

use transifex please

**Comment by MiquelRForgeFlow** [comment]
please @rafaelbn remove `duplicate` tag

@pedrobaeza all changes done plus some improvements

**Comment by MiquelRForgeFlow** [comment]
:green_apple:

File: product_supplierinfo_for_customer/README.rst
Comment (tarteo) [reply product_supplierinfo_for_customer/README.rst:1]: Ok, then we'll wait for the cron when it's merged.

**Pull request merged**
